### PR TITLE
feat(Heading/react): add component

### PR DESCRIPTION
**Pull request by krishnaacc123**
Add component

### CODE REVIEW

File: src/atoms/Heading/Heading.web.js [hunk 1]
```diff
@@ -1,5 +1,150 @@
...
+      case 'xxlarge':
+        return theme.fonts.size.xxxlarge;
+      case 'xxxlarge':
+        return theme.fonts.size.xxxxlarge;
+      default:
+        return theme.fonts.size.xxxxlarge;
+    }
+  },
+  color({ theme, color }) {
+    return getColor(theme, color);
```
Comment (deorukhkarchaitanya) [inline src/atoms/Heading/Heading.web.js:28]: You can do this inline

File: src/atoms/Heading/Heading.web.js [hunk 2]
```diff
@@ -1,5 +1,150 @@
...
+    default:
+      StyledHeading = StyledHeading2;
+  }
+  return (
+    <StyledHeading
+      size={size}
+      color={color}
+      numberOfLines={maxLines}
+      weight={weight}
+      data-testid={testID}
```
Comment (deorukhkarchaitanya) [inline src/atoms/Heading/Heading.web.js:126]: Use ```automation-attributes```

File: src/atoms/Heading/Heading.web.js [hunk 3]
```diff
@@ -1,5 +1,150 @@
...
+  color: ${styles.color};
+  letter-spacing: 0;
+  line-height: ${styles.lineHeight};
+`;
+
+const StyledHeading6 = styled.h6`
+  font-family: ${styles.fontFamily};
+  font-size: ${styles.fontSize};
+  color: ${styles.color};
+  letter-spacing: 0;
```
Comment (deorukhkarchaitanya) [inline src/atoms/Heading/Heading.web.js:92]: Some of the headings have letter-spacing, no? Can you check figma once?

File: src/atoms/Heading/__tests__/Heading.web.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Heading from '../index';
+import { renderWithTheme } from '../../../_helpers/testing';
+
+describe('<Heading />', () => {
+  describe('size', () => {
+    it('renders heading with size medium', () => {
```
Comment (dhruvdutt) [inline src/atoms/Heading/__tests__/Heading.web.test.js:7]: Add one test for default case.

File: src/atoms/Heading/Heading.web.js (hunk 3 above)
Comment (dhruvdutt) [reply src/atoms/Heading/Heading.web.js:92]: Yes, in one case it's there.

File: src/atoms/Heading/Heading.web.js [hunk 5]
```diff
@@ -1,5 +1,150 @@
...
+  children: PropTypes.string,
+  size: PropTypes.oneOf(['medium', 'large', 'xlarge', 'xxlarge', 'xxxlarge']),
+  color: PropTypes.oneOf(getColorKeys()),
+  testID: PropTypes.string,
+  maxLines: PropTypes.number,
+  weight: PropTypes.oneOf(Object.keys(baseTheme.fonts.weight)),
+};
+
+Heading.defaultProps = {
+  size: 'xxxlarge',
```
Comment (dhruvdutt) [inline src/atoms/Heading/Heading.web.js:143]: How are we deciding the default props?

File: src/atoms/Heading/Heading.web.js (hunk 2 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js:126]: added.

**Comment by krishnaacc123** [comment]
I've changed a log of logic and also added the feedback

File: src/atoms/Heading/__tests__/Heading.web.test.js (hunk 4 above)
Comment (krishnaacc123) [reply src/atoms/Heading/__tests__/Heading.web.test.js:12]: added

File: src/atoms/Heading/Heading.web.js [hunk 6]
```diff
@@ -1,5 +1,111 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { getColorKeys, getColor } from '../../_helpers/theme';
+import baseTheme from '../../tokens/theme';
+import automation from '../../_helpers/automation-attributes.web';
```
Comment (krishnaacc123) [inline src/atoms/Heading/Heading.web.js:6]: will change `automation-attributes.web` to `automation-attributes` this after the issue is resolved.

File: src/atoms/Heading/Heading.web.js (hunk 3 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js]: added

File: src/atoms/Heading/Heading.web.js (hunk 1 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js]: moved inline

File: src/atoms/Heading/Heading.web.js [hunk 7]
```diff
@@ -1,5 +1,108 @@
...
+      {children}
+    </StyledHeading>
+  );
+};
+
+Heading.propTypes = {
+  children: PropTypes.string,
+  size: PropTypes.oneOf(['medium', 'large', 'xlarge', 'xxlarge', 'xxxlarge']),
+  color: PropTypes.oneOf(getColorKeys()),
+  testID: PropTypes.string,
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:95]: we don't need testID same reason as I mentioned in Icon

File: src/atoms/Heading/Heading.web.js [hunk 8]
```diff
@@ -1,5 +1,108 @@
...
+    font-weight: ${({ theme, _weight }) => theme.fonts.weight[_weight]};
+    color: ${({ theme, color }) => getColor(theme, color)};
+    letter-spacing: 0;
+    line-height: ${styles.lineHeight};
+  }
+`;
+
+const Heading = ({ size, testID, color, children, maxLines, weight }) => {
+  let headingLevel;
+  switch (size) {
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:58]: just create an object outside this component with a map of size to level
```js
const headingLevel = {
'xxxlarge': 'h1',
'xxlarge': 'h2',
}
```
and so on. then you just need to refer it inside the component as 
`as={headingLevel[size]}`

File: src/atoms/Heading/Heading.web.js [hunk 9]
```diff
@@ -1,5 +1,108 @@
...
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+};
+
+const StyledHeading = styled(Text)`
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:45]:
```suggestion
const Heading = styled(Text)`
```

File: src/atoms/Heading/Heading.web.js [hunk 10]
```diff
@@ -1,5 +1,108 @@
...
+  maxLines: PropTypes.number,
+  weight: PropTypes.oneOf(Object.keys(baseTheme.fonts.weight)),
+};
+
+Heading.defaultProps = {
+  size: 'xxxlarge',
+  color: 'shade.980',
+  testID: 'ds-heading',
+  weight: 'bold',
+  maxLines: undefined,
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:105]: For Headings we should default `maxLines` to 1

File: src/atoms/Heading/__tests__/Heading.web.test.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import Heading from '../index';
+import { renderWithTheme } from '../../../_helpers/testing';
+
+describe('<Heading />', () => {
+  it('renders heading with no props', () => {
```
Comment (kamleshchandnani) [inline src/atoms/Heading/__tests__/Heading.web.test.js:6]:
```suggestion
  it('renders heading with default props', () => {
```

File: src/atoms/Heading/Heading.web.js (hunk 8 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js:58]: But then I'll need to change the size to support the `l_regular`,m_bold etc... but then native has only  support for basic five `sizes`.

Comment (kamleshchandnani) [reply src/atoms/Heading/Heading.web.js:58]: Aree nai nai my bad.
```
const headingLevel = {
  xxxl: H1,
  xxl: H2,
  xl: H3,
  L: H4,
  M: H5,
};
```

File: src/atoms/Heading/Heading.web.js [hunk 12]
```diff
@@ -1,5 +1,137 @@
...
+    font-weight: ${({ theme, weight }) => theme.fonts.weight[weight]};
+    color: ${({ theme, color }) => getColor(theme, color)};
+    letter-spacing: 0;
+    line-height: ${styles.lineHeight};
+    overflow: hidden;
+    text-overflow: ellipsis;
+    max-height: ${styles.maxHeight};
+    position: relative;
+    padding-right: 1em;
+    &&:before {
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:80]: Please don't do this. It should extend from Text. Even though text is broken right now it needs to be fixed. That's the whole point of a design system. The base should be fixed so all the components automatically gets fixed

File: src/atoms/Heading/Heading.web.js (hunk 9 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js:45]: At last I'm importing it as `Heading` only. 
Screenshot 2020-04-12 at 2 06 24 PM

File: src/atoms/Heading/Heading.web.js [hunk 13]
```diff
@@ -1,5 +1,95 @@
...
+  color: PropTypes.oneOf(getColorKeys()),
+  maxLines: PropTypes.number,
+  weight: PropTypes.oneOf(Object.keys(baseTheme.fonts.weight)),
+};
+
+Heading.defaultProps = {
+  size: 'xxxlarge',
+  color: 'shade.980',
+  weight: 'bold',
+  maxLines: 1,
```
Comment (deorukhkarchaitanya) [inline src/atoms/Heading/Heading.web.js:92]: maxLines should be undefined by default. Just like for Text

File: src/atoms/Heading/Heading.web.js (hunk 10 above)
Comment (dhruvdutt) [reply src/atoms/Heading/Heading.web.js:95]: Still pending?

Comment (dhruvdutt) [reply src/atoms/Heading/Heading.web.js:95]: @deorukhkarchaitanya Do we need to change this for native as well?

File: src/atoms/Heading/Heading.web.js [hunk 14]
```diff
@@ -1,5 +1,98 @@
...
+};
+
+const Heading = ({ size, color, children, maxLines, weight }) => {
+  return (
+    <StyledHeading
+      as={headingLevel[size]}
+      size={size}
+      color={color}
+      maxLines={maxLines}
+      _weight={weight}
```
Comment (dhruvdutt) [inline src/atoms/Heading/Heading.web.js:74]: Why adding `_`? It's a public prop anyway.

Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js:74]: Text has `_weight` prop.

Comment (dhruvdutt) [reply src/atoms/Heading/Heading.web.js:74]: Okay, got it.

File: src/atoms/Heading/Heading.web.js [hunk 15]
```diff
@@ -1,5 +1,98 @@
...
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+};
+
+const StyledHeading = styled(Text)`
```
Comment (dhruvdutt) [inline src/atoms/Heading/Heading.web.js:48]:
```suggestion
const StyledHeadingText = styled(Text)`
```

File: src/atoms/Heading/Heading.web.js [hunk 16]
```diff
@@ -1,5 +1,98 @@
...
+  xxxlarge: 'h1',
+  xxlarge: 'h2',
+  xlarge: 'h3',
+  large: 'h4',
+  medium: 'h5',
+};
+
+const Heading = ({ size, color, children, maxLines, weight }) => {
+  return (
+    <StyledHeading
```
Comment (dhruvdutt) [inline src/atoms/Heading/Heading.web.js:69]:
```suggestion
    <StyledHeadingText
```

File: src/atoms/Heading/Heading.web.js (hunk 10 above)
Comment (deorukhkarchaitanya) [reply src/atoms/Heading/Heading.web.js:95]: Yeah we should. I'll make that change.

File: src/atoms/Heading/Heading.web.js (hunk 16 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js]: added

File: src/atoms/Heading/Heading.web.js (hunk 15 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js]: added

File: src/atoms/Heading/Heading.web.js (hunk 12 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js]: Using Text

File: src/atoms/Heading/__tests__/Heading.web.test.js (hunk 11 above)
Comment (krishnaacc123) [reply src/atoms/Heading/__tests__/Heading.web.test.js]: replaced

File: src/atoms/Heading/Heading.web.js [hunk 17]
```diff
@@ -1,5 +1,96 @@
...
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+};
+
+const StyledHeadingText = styled(Text)`
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:48]: So finally we are not using the text component directly?

File: src/atoms/Heading/Heading.web.js [hunk 18]
```diff
@@ -1,5 +1,96 @@
...
+  &&& {
+    font-size: ${styles.fontSize};
+    color: ${({ theme, color }) => getColor(theme, color)};
+    letter-spacing: 0;
```
Comment (kamleshchandnani) [inline src/atoms/Heading/Heading.web.js:52]: we have all these props in Text

File: src/atoms/Heading/Heading.web.js (hunk 17 above)
Comment (krishnaacc123) [reply src/atoms/Heading/Heading.web.js]: Sorry about this. Actually Didn't really get this before when you were asking to change this.
I thought that you were asking to change the name of the component here.

File: src/atoms/Heading/Heading.native.js [hunk 19]
```diff
@@ -1,85 +1,77 @@
...
 const styles = {
-  fontFamily({ theme, weight }) {
-    return theme.fonts.family.lato[weight];
-  },
-  fontSize({ theme, size }) {
+  fontSize({ size }) {
     switch (size) {
       case 'medium':
-        return theme.fonts.size.medium;
+        return 'medium';
```
Comment (dhruvdutt) [inline src/atoms/Heading/Heading.native.js:13]: Have you tested native changes?

Comment (dhruvdutt) [reply src/atoms/Heading/Heading.native.js:13]: Okay, got it. Kamlesh has done it.

File: src/atoms/Heading/Heading.native.js [hunk 20]
```diff
@@ -1,85 +1,77 @@
...
       {children}
-    </StyledHeading>
+    </Text>
   );
 };
 
 Heading.propTypes = {
   children: PropTypes.string,
-  size: PropTypes.oneOf(['medium', 'large', 'xlarge', 'xxlarge', 'xxxlarge']),
+  size: PropTypes.oneOf(['medium', 'large', 'xlarge', 'xxlarge', 'xxxlarge']).isRequired,
```
Comment (deorukhkarchaitanya) [inline src/atoms/Heading/Heading.native.js:67]: Why is this required? Shouldn't it have a default behavior like Text does?

File: src/atoms/Heading/Heading.native.js [hunk 21]
```diff
@@ -1,85 +1,77 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import styled from 'styled-components/native';
-import { Text as NativeText } from 'react-native';
-import { getColorKeys, getColor } from '../../_helpers/theme';
+import { useTheme } from 'styled-components';
```
Comment (deorukhkarchaitanya) [inline src/atoms/Heading/Heading.native.js:3]: Just commit this directly here

```suggestion
import { useTheme } from 'styled-components/native';
```

**Comment by deorukhkarchaitanya** [comment]
@kamleshchandnani If we're adding more sizes to Text component, should we also include it in `Text.stories.js`?

File: src/atoms/Heading/Heading.native.js (hunk 20 above)
Comment (kamleshchandnani) [reply src/atoms/Heading/Heading.native.js:67]: Nupe Heading needs to have a size. Because on web H1, H2, H3 are different elements being rendered so it's basically to cautiously inform consumers to pick
